qcodes/instrument_drivers/stanford_research/SR830.py

```diff
@@ -695,3 +696,27 @@ def _set_sensitivity(self, s: float) -> int:
             return self._VOLT_TO_N[s]
         else:
             return self._CURR_TO_N[s]
+
+    def autorange(self, max_changes: int=1) -> None:
+        """
+        Automatically changes the sensitivity of the instrument according
+        to the R value and defined max_changes.
+        Args:
+            max_changes: Maximum number of steps allowing the function to
+            automatically change the sensitivity (default is 1). Actual
+            steps before reaching the limit in the function might be more
+            or less than this maximum.
```

Comment:
How can it be more ?

Reply:
Depends on the R value and the autorange implementation here, the sensitivity can change a finite number of steps before reaching to the limit that it doesn't change. So if you limit the max_changes steps below that headroom, it means that the actual change is more than what you set here by this max_changes.

Reply:
Ok so perhaps what you mean is `the actual number of steps needed to change to the optimal sensitivity may be more or less than the maximum`?

The number of steps performed is always less than or equal to the maximum right?

Reply:
> Ok so perhaps what you mean
Yes, it is what I mean. I will rephrase it.

> The number of steps performed
True